skyportal/handlers/api/followup_request.py

```diff
@@ -0,0 +1,84 @@
+import arrow
+from marshmallow.exceptions import ValidationError
+
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import DBSession, Instrument, Source, FollowupRequest, Token
+
+
+class FollowupRequestHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Submit follow-up request.
+        requestBody:
+          content:
+            application/json:
+              schema: FollowupRequest
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - Success
+                    - type: object
+                      properties:
+                        id:
+                          type: string
+                          description: New follow-up request ID
+        """
+        data = self.get_json()
+        _ = Source.get_if_owned_by(data["source_id"], self.current_user)
+        data["start_date"] = arrow.get(data["start_date"]).datetime
+        data["end_date"] = arrow.get(data["end_date"]).datetime
+        data["requester_id"] = self.current_user.id
+        if isinstance(data["filters"], str):
+            data["filters"] = [data["filters"]]
```

Comment:
Should be catching errors here -- for instance if the `start_date` is improperly formatted (ie. arrow throws an error)

Reply:
@profjsb if the parse fails, it automatically returns with an error code & message such that a notification is displayed on the front-end with the arrow exception message: `Backend error: Could not match input 'mydate' to any of the following formats: YYYY-MM-DD, YYYY-M-DD, YYYY-M-D, YYYY/MM/DD, YYYY/M/DD, YYYY/M/D, YYYY.MM.DD, YYYY.M.DD, YYYY.M.D, YYYYMMDD, YYYY-DDDD, YYYYDDDD, YYYY-MM, YYYY/MM, YYYY.MM, YYYY`. Does that suffice?

Reply:
Sure.

Reply:
As noted in the meeting earlier today, we should catch all errors so they don't get thrown to the backend logs.